console-subscriber/src/visitors.rs

```diff
@@ -0,0 +1,359 @@
+
+    const UPDATE_OP_ADD: &'static str = "add";
+    const UPDATE_OP_SUB: &'static str = "sub";
+    const UPDATE_OP_OVR: &'static str = "ovr";
+
+    pub(crate) fn result(self) -> Option<ResourceOpData> {
+        if let Some(op_name) = self.op_name {
+            if let Some(op_type) = self.op_type {
+                if op_type == Self::OP_TYPE_POLL {
+                    if let Some(readiness) = self.readiness {
+                        return Some(ResourceOpData::Poll { op_name, readiness });
+                    }
+                    return None;
+                } else if op_type == Self::OP_TYPE_STATE_UPDATE {
+                    let numeric_updates =
+                        self.state_numeric_attrs
+                            .into_iter()
+                            .filter_map(|(attr_name, attr)| {
+                                if let NumericStateAttr {
+                                    val: Some(val),
+                                    op: Some(op),
+                                    unit: Some(unit),
+                                } = attr
+                                {
+                                    let op = match op.as_str() {
+                                        Self::UPDATE_OP_ADD => Some(AttributeUpdateOp::Add),
+                                        Self::UPDATE_OP_SUB => Some(AttributeUpdateOp::Sub),
+                                        Self::UPDATE_OP_OVR => Some(AttributeUpdateOp::Ovr),
+                                        _ => None,
+                                    };
+
+                                    return op.map(|op| AttributeUpdate {
+                                        name: attr_name,
+                                        val: AttributeUpdateValue::Numeric { val, op, unit },
+                                    });
+                                }
+                                None
+                            });
+
+                    let text_updates =
+                        self.state_text_attrs.into_iter().map(|(attr_name, attr)| {
+                            AttributeUpdate {
+                                name: attr_name,
+                                val: AttributeUpdateValue::Text(attr),
+                            }
+                        });
+
+                    let attrs = numeric_updates.chain(text_updates).collect();
+                    return Some(ResourceOpData::StateUpdate { op_name, attrs });
+                }
+            }
+        }
+        None
+    }
+}
+
+struct MatchPart {
+    attr_name: String,
+    part_type: MatchPartType,
+}
+
+enum MatchPartType {
+    OpType(String),
+    Unit(String),
+    Text(String),
+}
+
+impl Visit for ResourceOpVisitor {
+    fn record_debug(&mut self, _: &field::Field, _: &dyn std::fmt::Debug) {}
+
+    fn record_str(&mut self, field: &tracing_core::Field, value: &str) {
+        let extract_partial_attr = || -> Option<MatchPart> {
+            let mut parts = field.name().split('_');
```

Comment:
ideally, we would want to use `.` instead of `_` but the tracing macro had some problem and did not allow me to emit events that have fields that are bot valid and invalid rust identifiers (the invalid ones being quoted `""`)

Reply:
Will put together an example shortly